python-threatexchange/threatexchange/exchanges/signal_exchange_api.py

```diff
@@ -169,19 +169,46 @@ def naive_fetch_merge(
                 old[k] = new_v
 
     @classmethod
-    @abstractmethod
     def naive_convert_to_signal_type(
         cls,
         signal_types: t.Sequence[t.Type[SignalType]],
         collab: TCollabConfig,
         fetched: t.Mapping[state.TUpdateRecordKey, state.TUpdateRecordValue],
-    ) -> t.Dict[t.Type[SignalType], t.Dict[str, state.TFetchedSignalMetadata]]:
+    ) -> t.Mapping[t.Type[SignalType], t.Dict[str, state.TFetchedSignalMetadata]]:
         """
         Convert the record from the API format to the format needed for indexing.
 
         This is the fallback method of creating state when there isn't a
         specialized storage for the fetch type.
+        """
+        patch = cls.naive_convert_to_signal_type_patch(signal_types, collab, fetched)
+        return patch[0]
 
+    @classmethod
+    @abstractmethod
+    def naive_convert_to_signal_type_patch(
+        cls,
+        signal_types: t.Sequence[t.Type[SignalType]],
+        collab: TCollabConfig,
+        fetched: t.Mapping[state.TUpdateRecordKey, state.TUpdateRecordValue],
+    ) -> t.Tuple[
+        t.Mapping[t.Type[SignalType], t.Dict[str, state.TFetchedSignalMetadata]],
+        t.Sequence[state.TUpdateRecordKey],
+    ]:
+        """
+        Convert records from the API's format to a format needed for patching to
+        a persistent storage.
```

Comment:
Hmm, this comment doesn't match my intent of how the persistent storage interface works. This interface suggests that the persistent storage should be k => <black box>, but for HMA you have redefined storage as needing to be (str, str) => metadata.

I am still convinced that the best way to have fetch API storage correct is to store it in the same native form that the APIs themselves support, which tends to vary, hence a keyspace specific to that API. Any other remapping introduces edge cases that require refcounting based on that key to be correct.

I think this interface is attempting to push the requirements for refcounting back into the SignalExchangeAPI, where it will then need to be repeatedly independently implemented.